WebDriverAgentLib/Categories/XCUIElement+FBFind.m

```diff
@@ -83,18 +85,42 @@ - (void)descendantsWithProperty:(NSString *)property value:(NSString *)value par
   // XPath will try to match elements only class name, so requesting elements by XCUIElementTypeAny will not work. We should use '*' instead.
   xpathQuery = [xpathQuery stringByReplacingOccurrencesOfString:@"XCUIElementTypeAny" withString:@"*"];
   NSArray *matchingSnapshots = [self.lastSnapshot fb_descendantsMatchingXPathQuery:xpathQuery];
-  NSArray *allElements = [[self descendantsMatchingType:XCUIElementTypeAny] allElementsBoundByIndex];
-  NSArray *matchingElements = [self filterElements:allElements matchingSnapshots:matchingSnapshots];
+  if (nil == matchingSnapshots) {
+    return @[];
+  }
+  // Prefiltering elements speeds up search by XPath a lot, because [element resolve] is the most expensive operation here
```

Comment:
The difference is visible on bigger test data structures. Testing app is too small to see it :(